### PR TITLE
docs: add storybook as output

### PR DESCRIPTION
**Pull request by jmfrancois**
refactor: move all examples into a dedicated folder
keep the existing CMF example app
install all tools around storybook to build something nice
create and use react-storybook-cmf plugin

### CODE REVIEW

File: docs/storybook/iframe.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+<link rel="stylesheet" href="http://fontawesome.io/assets/font-awesome/css/font-awesome.css">
+
+      </head>
+      <body>
+        <div id="root"></div>
+        <div id="error-display"></div>
+        <script src="static/preview.bundle.js"></script>
+      </body>
+    </html>
+
```
Comment (talend-glorieux) [inline docs/storybook/iframe.html]: Are those generated by storybook?

File: docs/storybook/static/preview.bundle.js.map [hunk 2]
```diff
@@ -0,0 +1 @@
+{"version":3,"file":"static/preview.bundle.js","sources":["webpack:///static/preview.bundle.js","webpack:///"],"mappings":"AAAA;AC+0HA;AAkqIA;AA+4HA;AAs/EA;AA6gGA;AAu3EA;AA48EA;AA0oGA;AAy6HA;AA6gHA;AAipKA;AA2sCA;AAwpGA;AAwtFA;AAm1CA;AAyBA;AAixFA;AA6tEA;AA2yGA;AAo2FA;AAu1FA;AA4qFA;AAixFA;AAylGA;AA2oHA","sourceRoot":""}
```
Comment (talend-glorieux) [inline docs/storybook/static/preview.bundle.js.map]: I'm not too OK with committing map files. Those usually end up being huge.

File: .storybook/config.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import cmf from 'react-storybook-cmf';
+import infoAddon from '@kadira/react-storybook-addon-info';
+import examples from '../examples';
+
+setAddon(infoAddon);
+setAddon({ addWithCMF: cmf.addWithCMF });
+
+function loadStories() {
+	Object.keys(examples).forEach((example) => {
+		console.log('example ', example);
```
Comment (talend-glorieux) [inline .storybook/config.js]: Why the console.log?

File: docs/storybook/iframe.html (hunk 1 above)
Comment (jmfrancois) [inline docs/storybook/iframe.html]: yes

File: docs/storybook/static/preview.bundle.js.map (hunk 2 above)
Comment (jmfrancois) [inline docs/storybook/static/preview.bundle.js.map]: oh true, it should be in the gitignore

File: .storybook/config.js (hunk 3 above)
Comment (jmfrancois) [inline .storybook/config.js]: My fault. I will remove it

File: docs/index.html [hunk 4]
```diff
@@ -22,7 +22,8 @@
           <p>A set of widgets</p>
           <ul>
             <li><a href="jsdoc">JSDoc</a></li>
-            <li><a href="examples/build">Examples</a></li>
+            <li><a href="examples/build">Examples throw react-cmf</a></li>
```
Comment (jsomsanith) [inline docs/index.html]: Do you mean 'through' ?
(same for the next line)

**Comment by jmfrancois** [comment]
I have done some changes following your reviews.
I have added npm start support to open storybook and work with it

**Pull request merged**